lib/ClangImporter/ClangIncludePaths.cpp

```diff
@@ -33,42 +33,64 @@ static std::optional<Path> getActualModuleMapPath(
-
-  StringRef SDKPath = Opts.getSDKPath();
-  if (!SDKPath.empty()) {
-    result.append(SDKPath.begin(), SDKPath.end());
-    llvm::sys::path::append(result, "usr", "lib", "swift");
-    llvm::sys::path::append(result, platform);
-    if (isArchSpecific) {
-      llvm::sys::path::append(result, arch);
+  auto checkRuntimeResourcesPath = [&]() -> std::optional<Path> {
+    Path result;
+    if (!Opts.RuntimeResourcePath.empty()) {
+      result.append(Opts.RuntimeResourcePath.begin(),
+                    Opts.RuntimeResourcePath.end());
+      llvm::sys::path::append(result, platform);
+      if (isArchSpecific) {
+        llvm::sys::path::append(result, arch);
+      }
+      llvm::sys::path::append(result, name);
+
+      // Only specify the module map if that file actually exists. It may not;
+      // for example in the case that `swiftc -target x86_64-unknown-linux-gnu
+      // -emit-ir` is invoked using a Swift compiler not built for Linux
+      // targets.
+      if (vfs->exists(result))
+        return result;
     }
-    llvm::sys::path::append(result, name);
-
-    // Only specify the module map if that file actually exists.  It may not;
-    // for example in the case that `swiftc -target x86_64-unknown-linux-gnu
-    // -emit-ir` is invoked using a Swift compiler not built for Linux targets.
-    if (vfs->exists(result))
-      return result;
-  }
+    return std::nullopt;
+  };
 
-  if (!Opts.RuntimeResourcePath.empty()) {
-    result.clear();
-    result.append(Opts.RuntimeResourcePath.begin(),
-                  Opts.RuntimeResourcePath.end());
-    llvm::sys::path::append(result, platform);
-    if (isArchSpecific) {
-      llvm::sys::path::append(result, arch);
+  auto checkSDKPath = [&]() -> std::optional<Path> {
+    Path result;
+    StringRef SDKPath = Opts.getSDKPath();
+    if (!SDKPath.empty()) {
+      result.append(SDKPath.begin(), SDKPath.end());
+      llvm::sys::path::append(result, "usr", "lib", "swift");
+      llvm::sys::path::append(result, platform);
+      if (isArchSpecific) {
+        llvm::sys::path::append(result, arch);
+      }
+      llvm::sys::path::append(result, name);
+
+      // Only specify the module map if that file actually exists.  It may not;
+      // for example in the case that `swiftc -target x86_64-unknown-linux-gnu
+      // -emit-ir` is invoked using a Swift compiler not built for Linux
+      // targets.
+      if (vfs->exists(result))
+        return result;
     }
-    llvm::sys::path::append(result, name);
+    return std::nullopt;
+  };
 
-    // Only specify the module map if that file actually exists.  It may not;
-    // for example in the case that `swiftc -target x86_64-unknown-linux-gnu
-    // -emit-ir` is invoked using a Swift compiler not built for Linux targets.
-    if (vfs->exists(result))
-      return result;
+  // FIXME: This is a workaround to keep the Android SDK build working.
+  // See https://github.com/swiftlang/swift/pull/74814.
+  if (triple.isAndroid()) {
```

Comment:
Put this Android block inside a `#if defined(_WIN32) || defined(_WIN64)`, as only The Browser Company's upcoming Android SDK that is only used on Windows and uses a new experimental SDK layout fails with this pull.

Reply:
I don't know if that's the best approach, I'm curious what @compnerd @hyp think is the best solution here.